google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/ConnectionWorker.java

```diff
@@ -278,6 +303,83 @@ static String getRoutingHeader(String streamName, String location) {
     return project + "locations/" + location;
   }
 
+  private Attributes buildOpenTelemetryAttributes() {
+    AttributesBuilder builder = Attributes.builder().put(telemetryKeyStreamId, this.streamName);
+    builder.put(telemetryKeyWriterId, this.writerId);
+    if ((this.traceId != null) && !this.traceId.isEmpty()) {
+      String[] traceIdParts = this.traceId.split(":", 5);
+      for (int i = 0; i < traceIdParts.length; i++) {
+        builder.put(telemetryKeysTraceId.get(i), traceIdParts[i]);
+      }
+    }
+    return builder.build();
+  }
+
+  @VisibleForTesting
+  Attributes getTelemetryAttributes() {
+    // streamName can change due to multiplexing. If it has changed, update the attributes.
+    String originalStreamNameAttribute = telemetryAttributes.get(telemetryKeyStreamId);
+    if ((originalStreamNameAttribute != null)
+        && !originalStreamNameAttribute.equals(this.streamName)) {
+      AttributesBuilder builder = telemetryAttributes.toBuilder();
+      builder.put("streamId", this.streamName);
+      telemetryAttributes = builder.build();
+    }
+    return telemetryAttributes;
+  }
+
+  private void registerOpenTelemetryMetrics() {
+    MeterProvider meterProvider = Singletons.getOpenTelemetry().getMeterProvider();
+    writeMeter =
+        meterProvider
+            .meterBuilder("com.google.cloud.bigquery.storage.v1.write")
+            .setInstrumentationVersion(
+                ConnectionWorker.class.getPackage().getImplementationVersion())
+            .build();
+    instrumentIncomingRequestCount =
+        writeMeter
+            .counterBuilder("incoming-request-count")
+            .setDescription("Counts number of incoming requests")
+            .build();
+    instrumentIncomingRequestSize =
+        writeMeter
+            .counterBuilder("incoming-request-size")
+            .setDescription("Counts byte size of incoming requests")
+            .build();
+    writeMeter
```

Comment:
This seems to be more java client lib specific metric, I am wondering if we should have a sub-namespace for it.

Reply:
I have renamed the 3 metrics to be consistent with the Go client lib. Now these are:

append_requests
append_request_bytes
append_rows

I also removed the metrics related to length of waiting and inflight queues for now.